macros/src/types/type_as.rs

```diff
@@ -0,0 +1,65 @@
+use quote::quote;
+use syn::{Result, Type};
+
+use crate::{
+    attr::{EnumAttr, StructAttr},
+    deps::Dependencies,
+    DerivedTS,
+};
+
+pub(crate) fn type_as_struct(attr: &StructAttr, name: &str, type_as: &Type) -> Result<DerivedTS> {
+    if attr.rename_all.is_some() {
+        syn_err!("`rename_all` is not compatible with `as`");
+    }
+    if attr.tag.is_some() {
+        syn_err!("`tag` is not compatible with `as`");
+    }
```

Comment:
In both this and the type_override file, make sure `type` and `as` also emit a not compatible error. Other than that, LGTM